services/report/__init__.py

```diff
@@ -986,6 +986,21 @@ def build_report_from_raw_content(
                 is_retryable=True,
             )
             return result
+        except Exception as e:
+            log.exception(
+                "Unknown error when fetching raw report from storage",
+                extra=dict(
+                    repoid=commit.repoid,
+                    commit=commit.commitid,
+                    archive_path=archive_url,
+                ),
+            )
+            result.error = ProcessingError(
```

Comment:
Maybe not the scope of this PR, but do we intend to eventually set the upload's state also to some error? A state seems to also be "error", while we also have specific errors tied to an upload, so not sure what's the expected result here

Reply:
@adrian-codecov returning a ProcessingError is how the state gets uploaded correctly ([here](https://github.com/codecov/worker/blob/7ebdf9f9140a619b71347c4597d6d68079cb2850/services/report/__init__.py#L1040)). this PR basically plugs holes where we failed to return a ProcessingError when an error occurred

Reply:
That makes sense, ty for pointing it 👌